### PR TITLE
chore(sdk): making `api.vars` mandatory and returning an empty map by default

### LINKED ISSUE #2144: `req.vars` should not be optional
### I tried this

Use `req.vars`

### I expected this:

If there are no vars in API requests, the `req.vars` map should not be `nil` but rather an empty map.

### Instead, this happened

It is optional, so it was harder to use

### Is there a workaround?

Yes, something like:

```js
let vars = req.vars ?? Map<str>{};
```

### Component

SDK

### Wing Version

_No response_

### Wing Console Version

_No response_

### Node.js Version

_No response_

### Platform(s)

_No response_

### Anything else?

_No response_

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

### PR DESCRIPTION
**Pull request by garysassano**
Closes #2144

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

**Review by tsuf239 — COMMENTED** [review]
Great! Can you please:
1. change the code on `examples/tests/valid/api_path_vars.w`, 
2. run `npx nx test hangar` (to generate snapshots)
3. run `npm run build`

File: docs/04-reference/wingsdk-api.md [hunk 1]
```diff
@@ -1557,39 +1557,39 @@ The request's query string values.
 
 ---
 
-##### `body`<sup>Optional</sup> <a name="body" id="@winglang/sdk.cloud.ApiRequest.property.body"></a>
+##### `vars`<sup>Required</sup> <a name="vars" id="@winglang/sdk.cloud.ApiRequest.property.vars"></a>
```
Comment (staycoolcall911) [inline docs/04-reference/wingsdk-api.md:1560]: Hey @garysassano we would be grateful if you could also update the relevant part of [Wing SDK Spec](https://github.com/winglang/wing/blob/main/docs/04-reference/wingsdk-spec.md#api) as part of your PR 🙏

File: examples/tests/valid/api_path_vars.w [hunk 2]
```diff
@@ -7,7 +7,7 @@ class Fetch {
 }
 
 let handler = inflight (req: cloud.ApiRequest): cloud.ApiResponse => {
-  let vars = req.vars ?? {name: ""};
+  let vars = req.vars;
   return cloud.ApiResponse {
     body: {user: vars.get("name")},
```
Comment (tsuf239) [inline examples/tests/valid/api_path_vars.w:12]:
```suggestion
  return cloud.ApiResponse {
    body: {user:  req.vars.get("name")},
```

File: docs/04-reference/wingsdk-api.md (hunk 1 above)
Comment (garysassano) [reply docs/04-reference/wingsdk-api.md:1560]: Thanks a lot for your input, updated that as well.

**Review by tsuf239 — APPROVED** [review]
🏆

**Pull request merged**